cpp/src/db/db_execution.cu

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <db/db_execution.cuh>
+#include <fstream>
+#include <sstream>
+#include <string>
+
+namespace cugraph {
+namespace db {
+string_table& string_table::operator=(string_table&& other)
+{
+  this->names   = std::move(other.names);
+  this->columns = std::move(other.columns);
+  return *this;
+}
+
+string_table::string_table() {}
+
+string_table::string_table(string_table&& other)
+{
+  this->names   = std::move(other.names);
+  this->columns = std::move(other.columns);
+}
+
+string_table::string_table(std::string csvFile, bool with_headers, std::string delim)
+{
+  std::ifstream myFileStream(csvFile);
```

Comment:
Why are you writing your own CSV reader rather than just calling cudf.read_csv?

Reply:
Because the result needs to be on the host, and cudf.read_csv puts it on the GPU.